test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -6394,6 +6394,29 @@ BOOST_AUTO_TEST_CASE(inline_long_string_return)
 	compileAndRun(sourceCode, 0, "C");
 	BOOST_CHECK(callContractFunction("f()") == encodeDyn(strLong));
 }
+
+BOOST_AUTO_TEST_CASE(fixed_bytes_index_access)
+{
+	char const* sourceCode = R"(
+		contract C {
+			bytes16[] public data;
+			function f(bytes32 x) returns (byte) {
+				return x[2];
+			}
+			function g(bytes32 x) returns (uint) {
+				data = [x[0], x[1], x[2]];
+				data[0] = "12345";
+				return uint(data[0][4]);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("f(bytes32)", "789") == encodeArgs("9"));
+	BOOST_CHECK(callContractFunction("g(bytes32)", "789") == encodeArgs(u256(int('5'))));
+	BOOST_CHECK(callContractFunction("data(uint256)", u256(1)) == encodeArgs("8"));
```

Comment:
We cannot do it with string because "length" for a string is the number of characters. To obtain that, we would need to count them, which takes longer than the short `.length` would suggest.